# Visualising a Phylogenetic Tree { #sec-itol }

In this section you will use the online tool `iTOL` to visualise the phylogenetic tree you created in @sec-tree.
This tree represents evolutionary relationships between the 33 study isolates, as inferred from differences in their genome sequences.
By appropriately _rooting_ and annotating the tree, you should be able to draw conclusions concerning the routes of transmission of these isolates.

## Downloading the tree

To use `iTOL` you need to upload a file representing the phylogenetic tree.
At the moment your tree file only exists on `Galaxy` so you need to download it to your local project folder.
To download the tree file:

1. Click on the `Best-scoring ML Tree` history item
2. Click on the floppy disk icon[^1]
3. Select the appropriate location to download the file
4. Click `Save`

[^1]: We know you're all too young to have used a floppy disk, but humour us...

This will place the Newick-formatted tree file in your local directory.

::: { .callout-note }
## Video: Downloading the tree file

{{< video assets/movies/galaxy-23-tree.mp4 title="Downloading the tree file from `Galaxy`" >}}
:::

## Upload the tree to `iTOL`

1. Navigate to the `iTOL` website at [https://itol.embl.de/](https://itol.embl.de/)
2. Click on `Upload a tree`
3. Click on `Browse` and navigate to your downloaded Newick tree file
4. Select the tree file and click `Open`. The tree file's name should appear at the `iTOL` site
5. Click `Upload`

A basic representation of your tree should appear, as in @fig-itol-1

::: { .column-margin }

![Initial tree representation at `iTOL`](assets/images/itol-1.png){#fig-itol-1 width=80% .lightbox}

:::

::: { .callout-note }
## Video: Upload the tree file to `iTOL`

{{< video assets/movies/itol-01-upload.mp4 title="Uploading the tree file to `iTOL`." >}}
:::

## Re-root the tree

::: { .callout-important }
Your colleague has been super-helpful and identified the reference genome that is assumed to be ancestral to all of the genomes in the tree.
The ancestral genome, which should be considered the root of the tree, is `ERR531413_asm.fasta.ref`.
Rooting the tree at this genome should make the tree reflect the time sequence of divergence of each isolate from their common ancestors.
:::

1. Click on the ancestral node in the tree: `ERR531413_asm.fasta.ref`
2. Click on `Tree structure -> Re-root the tree here`

::: { .callout-note }
## Video: Re-root the tree to the known ancestral reference

{{< video assets/movies/itol-02-reroot.mp4 title="Re-rooting the tree." >}}
:::

::: { .callout-todo }
**ADD QUESTION ASKING STUDENTS TO CONSIDER THE CLADE STRUCTURE**
:::

