# Visualise the Mapping {#sec-jbrowse}

The `.bam` file generated in @sec-bwamem contains the information we need to understand how well the reads map to . However, being plain-text files and often quite large, they are not easy to read and process intuitively.

To get the most benefit out of these files, we usually visualise them using an appropriate software tool. The choice of tool depends on the reason for visualisation. The [`Artemis`](https://www.sanger.ac.uk/tool/artemis/) software (@Carver2012-ja) is useful for interactive manual genome annotation; [`Tablet`](https://ics.hutton.ac.uk/tablet/) is an excellent tool for investigating genome assembly and sequence variation (@Milne2013-vp)[^1]; and [`Proksee`](https://proksee.ca/) is an online service (@Grant2023-zr) often used to obtain representative images of genome assembly annotations.

[^1]: `Tablet` is the standalone tool I would prefer to use to visualise this data (_LP_).

- [`Artemis` documentation](https://sanger-pathogens.github.io/Artemis/Artemis/)
- [`Tablet` documentation](https://ics.hutton.ac.uk/tablet/)
- [`Proksee` service](https://proksee.ca/)

In this section, you will use [`JBrowse`](https://jbrowse.org/jb2/) (@Diesh2023-go) to visualise your reference mapping interactively. `JBrowse` is a lightweight genome browser written in JavaScript that can run as a standalone application, or embedded as a web tool, as it is in Galaxy.

## Visualising the Read Mapping

To use `JBrowse` to investigate your read mapping you need to tell it both what to display, and how to display it.

::: { .callout-warning }
There are a number of new concepts being introduced here, and they may not make sense until you have tried to visualise your data.

If you follow the instructions below, and watch the video, you should find that everything works, and the meaning of unfamiliar terms like _track group_ will become apparent.
:::

1. Navigate to the `JBrowse` tool in the `Tools` sidebar
2. Select the `JBrowse` tool
3. Under `Reference genome to display`, select the `Use a genome from history` option
4. Select the `GCF_000006765.1_ASM676v1_genomic.fna.gz` genome file in `Select the reference genome`
5. Click on `+ Insert Track Group` to add a new _track group_
6. In the new track group, enter `Mapping` into the `Track Category` field
7. Click on `+ Insert Annotation Track` to add a new track to hold the mapping data
8. Make sure that the `Track Type` of the new annotation track is `BAM Pileups`
9. In `BAM Track Data`, select the `BWA-MEM2` `.bam` output file
10. Select `Yes` under `Autogenerate SNP Track`
11. Click `Run Tool`

::: { .callout-todo }
**GENERATE VIDEO OF MAPPING VISUALISATION IN GALAXY**
:::

::: { .callout-note }
## Video: Configuring `JBrowse` to visualise `BWA-MEM2` mapping data

{{< video assets/movies/bwamem-3_full.mp4 title="Configuring `JBrowse` to visualise `BWA-MEM2` mapping data" >}}
:::

Click on the `eye` icon for the new `JBrowse` output, and select the following tracks to be shown:

1. Reference sequence
2. Mapping (`SNPs/Coverage`)

to obtain a visualisation of reads that map to the reference genome.

::: { .callout-todo }
**GENERATE IMAGE OF MAPPING VISUALISATION IN GALAXY**
:::

![`JBrowse` visualisation of a `BWA-MEM2` mapping of SARS-CoV-2 reads against a SARS 2003 reference genome. The annotation track shows predicted genes, including the spike (S) protein. The mapping track shows individual SNPs (teardrops), and total mapping read depth (the stacks above the teardrops).](assets/images/bwamem_mapping.png){#fig-bwamem-1 width="80%"}

You can click on the header of the SNP track in `JBrowse` and select `Pin to top`, so that you can zoom in and see individual SNPs, as in the video below.

::: { .callout-todo }
**GENERATE VIDEO OF SNP VISUALISATION IN GALAXY**
:::

::: { .callout-note }
## Video: Visualising SNPs in `JBrowse`

{{< video assets/movies/bwamem-4_full.mp4 title="Visualising SNPs in `JBrowse`" >}}
:::

::: { .callout-questions }
1. Do the `ERR31380` reads map uniformly to the reference genome?
2. Where do the reads appear to differ?
3. When `ERR31380` reads do map to the reference genome, how many SNPs are there?
:::

::: { .callout-challenge }
1. Visualise how your `ERR31380` reads map to the `ERR531413_asm` genome assembly.
2. How does this mapping differ from that to the _P. aeruginosa_ reference genome.
:::

## Next Steps

We can find SNPs by visual inspection in a tool like `JBrowse`, but there can be very many of them - as you can see. To ensure reproducibility and accuracy, and avoid errors we need to use dedicated software and pipelines when analysing mapping data. You will have a chance to do this in @sec-snippy.
