# Assemble the `ERR531380` Genome {#sec-shovill}

::: { .callout-important }
## Someone else has assembled the genome for you

The next step in the investigation would normally be to assemble the `ERR531380` genome and assess the assembly quality.

**Using `Galaxy`, genome assembly would take too long for this workshop.**

In this scenario, a colleague has kindly assembled your genome for you using the `SPAdes` assembler (@Prjibelski2020-iq) and provided the result for you to download in two files, below - the assembled genome (`.fasta` file), and the assembly graph (`.gfa` file):

- [`ERR531380.contigs.fasta`](assets/workshop_downloads/ERR531380.contigs.fasta)
- [`ERR531380.gfa`](assets/workshop_downloads/ERR531380.gfa)

1. Download the assembled genome and assembly graph to the local folder containing your workshop files
2. Upload both files to `Galaxy`

::: { .callout-note }
## Video: Upload the `ERR531380` assembly files

{{< video assets/movies/galaxy-17-upload.mp4 title="Uploading the `ERR531380` assembly files to Galaxy." >}}
:::

:::

In this section you will investigate the quality of this genome assembly visually using the `Bandage` tool. You will also use the `CheckM` tool to obtain a measure of genome completeness.

## Visualising the Assembly

The `Bandage` software package (@Wick2015-vm) can take assembly output from tools such as `SPAdes` and `Shovill`, and visualise them as a graph. This is a useful step in assessing the quality of an assembly and can help identify poorly-assembled regions and when there is potential unintentional sequencing of co-cultures of related strains, rather than an axenic isolate.

- [`Bandage` documentation](https://github.com/rrwick/Bandage/wiki)
- [`Bandage` website](https://rrwick.github.io/Bandage/)

To use `Bandage` to visualise your genome assembly

1. Navigate to the `Bandage Image` tool using the `Tools` sidebar in Galaxy
2. Select the `Bandage Image` tool
3. Make sure that you have selected the `.gfa` file as the `Graphical Fragment Assembly` input
4. Click `Run Tool`

::: { .callout-note }
## Video: Visualising the `ERR531380` assembly using `Bandage Image`

{{< video assets/movies/galaxy-18-bandage.mp4 title="Visualising the genome assembly with `Bandage Image`" >}}
:::

::: { .callout-tip }
**Inspect the assembly graph by clicking on the `eye` icon for the Assembly Graph Image output.**
:::

## Estimating Assembly Quality

The `CheckM` software (@Parks2015-so) is a suite of tools used to assess the quality of a bacterial assembly. It estimates the completeness of the genome by looking for the presence of _single copy marker genes_ (SCMGs) for a stated phylogenetic lineage. The more of these marker genes that can be determined to be present in the genome, the more complete the assembly is assumed to be.

- [`CheckM` docmentation](https://github.com/Ecogenomics/CheckM/wiki)
- [`CheckM` website](https://ecogenomics.github.io/CheckM/)

The use `CheckM` to estimate the quality of your assembly:

1. Navigate to the `CheckM taxonomy_wf` tool using the `Tools` sidebar in Galaxy
2. Select the Taxonomic rank of `Species` and set the `Taxon of interest` to be _Pseudomonas aeruginosa_ (the identity of the isolate is known, or at least strongly suspected, on the basis of polyphasic tests)
3. Under `Data structure for bins` select `In individual datasets` and choose the `ERR531380.contigs.fasta` file
4. Click on `Run Tool`

::: { .callout-note }
## Video: Estimating assembly quality using `CheckM`

{{< video assets/movies/galaxy-19-checkm.mp4 title="Estimating assembly quality with `CheckM`" >}}
:::

## Next steps

With the draft genome assembly you have for `ERR531380` you can use pubMLST to classify your isolate in more detail, using `Galaxy` and the `pubMLST` server.
